ipc_receivefile.c

```diff
@@ -0,0 +1,184 @@
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/ipc.h>
+#include <sys/msg.h>
+#include <mqueue.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <errno.h>
+
+/*The msg_len argument specifies the size of the buffer
+  pointed to by msg_ptr; this must be greater than or equal to the
+  mq_msgsize attribute of the queue*/
+       
+#define MQ_MAXMSG 10
+#define MQ_MSGSIZE 8192
+
+void write_to_file(char *data, size_t data_size, long int fd);
+long int find_file_size(long int fd);
+
+int main(int argc, char *argv[])
+{
+	int opt;
+	int fflag = 0;
+	int qflag = 0;
+	int pflag = 0;
+	int sflag = 0;
+	char method_name[50]={};
+	char file_name[50]={};
+	// file size
+	long int fd;	
+	long int send = 0;
+	
+	if (argc < 2)
+		printf("Missing arguments or extra arguments, use -h for more information.\n");
+
+
+	while ((opt = getopt(argc, argv, "hf:q:p:s:")) != -1) {
+		switch (opt) {
+			case 'h':
+				printf("<HELP>\n"
+					"-q <queue_name>, use queue as IPC method\n"
+					"-p <pipe_name>, uses pipes as IPC method\n"
+					"-s <buffer size in kB>, uses shared memory as IPC method\n"
+					"-f <filename>, file used to read/write data\n"
+					"</HELP>\n");
+			break;
+			case 'q':
+				qflag ++;
+				strcpy(method_name, optarg);
+				break;
+			case 'p':
+				pflag ++;
+				strcpy(method_name, optarg);
+				break;
+			case 's':
+				sflag ++;
+				strcpy(method_name, optarg);
+				break;
+			case 'f':
+				fflag ++;
+				strcpy(file_name, optarg);
+				fd = open(file_name, O_RDWR,0660);
```

Comment:
```suggestion
				fd = open(file_name, O_RDWR | O_CREAT | O_EXCL ,0660);
```
When receiving a file we expect the destination file not to exist

Reply:
You are right. I will change it as your suggestion